pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -186,6 +188,12 @@ protected void batchOperation(List<MetadataOp> ops) {
                     }
                 }
             }, null);
+
+            executor.schedule(() -> {
+                if (!callback.get()) {
+                    ops.forEach(n -> n.getFuture().completeExceptionally(new TimeoutException()));
+                }
+            }, getMetadataStoreConfig().getOperationTimeoutSeconds(), TimeUnit.SECONDS);
```

Comment:
I think we don't need to change here? The batch operation just groups a few ops to one batch, each single ops has a future and if the caller requires an operation timeout, the caller can only use `future.get(timeout)`?

Reply:
If the caller use `future.join()`, the caller will be blocked.

Reply:
Yes, using` future.join()` means the caller expects an infinite timeout or without a timeout. it's dangerous here to add operation timeout. If the broker acquired a lock from Zookeeper but the operation timeout happened first, the callback happened later, how do we deal with this case?

I think the main point is to find the root cause of why the the zk doesn't call the callback, is the performance bottleneck or deadlock?

Reply:
> If the broker acquired a lock from Zookeeper but the operation timeout happened first, the callback happened later, how do we deal with this case?

We cannot determine the order, maybe we should avoid using `future.join()` in our project, but I still care when the zk doesn't call the callback, we should how to release a future.

> I think the main point is to find the root cause of why the the zk doesn't call the callback, is the performance bottleneck or deadlock?

I have found the root cause of zk doesn't call the callback, and I submitted #13809 to fix this.